src/bun.js/node/node_fs.zig

```diff
@@ -3582,6 +3582,42 @@ pub const NodeFS = struct {
     pub fn rmdir(this: *NodeFS, args: Arguments.RmDir, comptime flavor: Flavor) Maybe(Return.Rmdir) {
         switch (comptime flavor) {
             .sync => {
+                if (args.recursive) {
+                    std.fs.cwd().deleteTree(args.path.slice()) catch |err| {
```

Comment:
Similar to `rm`, we should only use `deleteTree` on Linux. On macOS, we should use `removefile` and later we should rewrite deleteTree to not use `unreachable` in any of the functions

Reply:
updated to use `removefileat`